objects/src/assets/mod.rs

```diff
@@ -37,12 +37,32 @@ use core::{fmt, ops::Deref};
 ///   This guarantees that finding two assets with the same hash is infeasible.
 /// - Lastly, the first bit of the result is set to ZERO, and the last 32 bits of the 3rd element
 ///   are set to 2^31 (i.e., ONE followed by 32 ZEROs).
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub enum Asset {
     Fungible(FungibleAsset),
     NonFungible(NonFungibleAsset),
 }
 
+impl Asset {
+    /// Returns true if this asset is the same as the specified asset.
+    ///
+    /// Two assets are defined to be the same if:
+    /// - For fungible assets, if they were issued by the same faucet.
+    /// - For non-fungible assets, if the assets are identical.
+    pub fn is_same(&self, other: &Self) -> bool {
```

Comment:
It is not clearly distinguishable from `PartialEq` & `Eq` implementation. Do we want to allow `asset_a == asset_b` with this implementation? If no, then we should use the traits. If yes, then we probably want to rename the method to `is_equivalent`. Or, `has_same_id_than`

Reply:
This is a bit tricky because for non-fungible assets this checks for exact equality (i.e., `==`) but for fungible assets this checks only for equality of IDs. So, we can't use the traits at this level, and also something like `has_same_id` is not really correct. Regarding `is_same` vs. `is_equivalent` - I think they are both similar, but `is_same` is shorter and maybe a tiny bit more clear.